File: src/loadData.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <vector>
#include <sstream>
#include "loadData.h"

// function stod allows me to convert string to doubles
// the initial getline function throws out my header line
std::pair<std::vector<double>, std::vector<double>> storeData(std::ifstream& file) {
    std::vector<double> timeData;
    std::vector<double> signalData;
    std::string line;

    if(!file.is_open()) {
        std::cerr << "File does not exist\n";
        return {{}, {}};
    }
    std::getline(file, line); // skip header
    std::cout<<"Loading data from file\n";
    while(std::getline(file, line)) {
        std::stringstream ss(line);
        std::string timeStr, signalStr;
        if (std::getline(ss, timeStr, ',') && std::getline(ss, signalStr)) {
            double index = std::stod(timeStr);
            double time = index / 360.0; // Getting the time using the Hz
            double signal = std::stod(signalStr);
            timeData.push_back(time);
            signalData.push_back(signal);
        }
    }
    if (timeData.size() == signalData.size()) {
        std::cout << "Data Loaded\n";
    }
    return {timeData, signalData};
}

File: python/loadData.py
# This file is for loading data and can be used to add other functions to load other types of data
import numpy as np
import csv

def loadData(filename, frequency):
    """
          Loads ECG data from a CSV file.
          Args:
              filename: Path to the CSV file containing ECG data.
          Returns:
              A tuple of two lists (timeData, signalData).
          """
    timeD = []
    signalD = []
    with open(filename,  newline='') as csvfile:
        filereader = csv.reader(csvfile)
        next(filereader, None)  # skip header
        for line in filereader:
            if len(line) < 2:
                continue
            index = float(line[0])
            signal = float(line[1])
            time = index / frequency  # Convert sample index to time in seconds (fs = 360 Hz)
            timeD.append(time)
            signalD.append(signal)
        if len(timeD) == len(signalD):
            print("Data loaded.\n")
        return np.array(timeD), np.array(signalD)

